Twittimation/OAuthClient.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web;
+
+namespace Twittimation
+{
+    public class OAuthClient
+    {
+        private string _apiUrl;
+        private string _consumerKey;
+        private string _consumerKeySecret;
+        private string _accessToken;
+        private string _accessTokenSecret;
+        private HMACSHA1 _encryptor;
+        
+        public OAuthClient(string apiUrl, string consumerKey, string consumerKeySecret, string accessToken, string accessTokenSecret)
+        {
+            _apiUrl = apiUrl;
+            _consumerKey = consumerKey;
+            _consumerKeySecret = consumerKeySecret;
+            _accessToken = accessToken;
+            _accessTokenSecret = accessTokenSecret;
+            _encryptor = new HMACSHA1(new ASCIIEncoding().GetBytes(consumerKeySecret + "&" + accessTokenSecret));
+        }
+
+        public async Task<string> SendRequest(string relativeUrl, Dictionary<string, string> data)
```

Comment:
This should accept a generic parameter <T> for a Data Structure and internally convert it to a Dictionary. `Dictionary<string, string>` should not be used by domain code.